packages/design-system/src/scalars/components/number-field/number-field-validations.ts

```diff
@@ -31,67 +31,70 @@ export const validateNumericType =
         if (allowNegative) return true;
-        // Si llegue hasta aqui no permito negative
         if (isPositive) return true;
         return "Value must be a positive integer";
       }
 
       case "NegativeInt": {
-        if (!isInt) return "Value must be an integer";
+        if (!isInt) return "Value must be a negative integer";
         if (!allowNegative && Number(value) >= 0)
           return "Value must be a negative integer";
         if (isNegative) return true;
         return "Value must be a negative integer";
       }
 
       case "NonNegativeInt": {
-        if (!isInt) return "Value must be an integer";
+        if (!isInt) return "Value must be a non-negative integer";
         if (!allowNegative && isNegative)
           return "Value must be a non-negative integer";
         if (Number(value) >= 0) return true;
         return "Value must be a non-negative integer";
       }
 
       case "NonPositiveInt": {
-        if (!isInt) return "Value must be an integer";
+        if (!isInt) return "Value must be a non-positive integer";
         if (!allowNegative && Number(value) > 0)
           return "Value must be a non-positive integer";
         if (Number(value) <= 0) return true;
         return "Value must be a non-positive integer";
       }
 
       case "PositiveFloat": {
-        if (!isFloat(value)) return "Value must be a float";
+        if (!isFloat(value) && !isInt)
+          return "Value must be a positive float or integer";
         if (!allowNegative && parseFloat(value as string) <= 0)
-          return "Value must be a positive float";
+          return "Value must be a positive float or integer";
         if (parseFloat(value as string) > 0) return true;
-        return "Value must be a positive float";
+        return "Value must be a positive float or integer";
       }
 
       case "NegativeFloat": {
-        if (!isFloat(value)) return "Value must be a float";
+        if (!isFloat(value) && !isInt)
+          return "Value must be a negative float or integer";
         if (!allowNegative && parseFloat(value as string) >= 0)
-          return "Value must be a negative float";
+          return "Value must be a negative float or integer";
         if (parseFloat(value as string) < 0) return true;
-        return "Value must be a negative float";
+        return "Value must be a negative float or integer";
       }
 
       case "NonNegativeFloat": {
-        if (!isFloat(value)) return "Value must be a float";
+        if (!isFloat(value) && !isInt)
+          return "Value must be a non-negative float or integer";
         if (!allowNegative && parseFloat(value as string) < 0)
-          return "Value must be a non-negative float";
+          return "Value must be a non-negative float or integer";
         if (parseFloat(value as string) >= 0) return true;
-        return "Value must be a non-negative float";
+        return "Value must be a non-negative float or integer";
       }
```

Comment:
This looks to be the same of `PositiveFloat`, you can avoid duplication to put the two cases together

Reply:
not applicable, for that case its handle different use cases